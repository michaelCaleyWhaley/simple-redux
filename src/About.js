import React from "react";
import { connect } from "react-redux";
import { updateName, updateEmail } from "./redux/actions";

function About({ name, email, dispatchUpdateName, dispatchUpdateEmail }) {
  return (
    <>
      <h2>About</h2>
      <form>
        <label for="name">Name:</label>
        <input
          type="text"
          id="name"
          name="name"
          value={name}
          onChange={(e) => {
            dispatchUpdateName(e.target.value);
          }}
        />

        <br />

        <label for="email">Email:</label>
        <input
          type="text"
          id="email"
          name="email"
          value={email}
          onChange={(e) => {
            dispatchUpdateEmail(e.target.value);
          }}
        />
      </form>
    </>
  );
}

const mapStateToProps = (state) => state;

const mapDispatchToProps = (dispatch) => ({
  dispatchUpdateName: (text) => {
    dispatch(updateName(text));
  },
  dispatchUpdateEmail: (text) => {
    dispatch(updateEmail(text));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(About);
