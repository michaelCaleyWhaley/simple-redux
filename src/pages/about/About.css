.heading {
  color: red;
}

.heading--alt {
  color: green;
}
