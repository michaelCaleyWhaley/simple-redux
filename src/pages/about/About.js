import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { connect } from "react-redux";
import { updateName, updateEmail } from "../../redux/actions";

import "./About.css";

function About({ name, email, dispatchUpdateName, dispatchUpdateEmail }) {
  const history = useHistory();
  function historyGoBack() {
    history.goBack();
  }

  const [h1Colour, setH1Colour] = useState(false);

  return (
    <>
      <h1
        data-testid="about-heading"
        className={h1Colour ? "heading" : "heading--alt"}
        onClick={() => {
          setH1Colour(!h1Colour);
        }}
      >
        About
      </h1>
      <form>
        <label>
          Name:
          <input
            data-testid="about-input"
            type="text"
            id="name"
            name="name"
            value={name}
            onChange={(e) => {
              dispatchUpdateName(e.target.value);
            }}
          />
        </label>

        <br />

        <label>
          Email:
          <input
            type="text"
            id="email"
            name="email"
            value={email}
            onChange={(e) => {
              dispatchUpdateEmail(e.target.value);
            }}
          />
        </label>
      </form>
      <button type="button" onClick={historyGoBack}>
        Back
      </button>
    </>
  );
}

const mapStateToProps = (state) => state;

const mapDispatchToProps = (dispatch) => ({
  dispatchUpdateName: (text) => {
    dispatch(updateName(text));
  },
  dispatchUpdateEmail: (text) => {
    dispatch(updateEmail(text));
  },
});

export { About };
export default connect(mapStateToProps, mapDispatchToProps)(About);
