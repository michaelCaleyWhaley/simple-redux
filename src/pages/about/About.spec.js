import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { About } from "./About";

let output;

const props = {
  name: "",
  email: "",
  dispatchUpdateName: jest.fn((n) => {
    output.value = output.value + n;
  }),
  dispatchUpdateEmail: jest.fn(),
};

describe("about page", () => {
  beforeEach(() => {
    output = { value: "" };
  });

  test("should render About component", () => {
    const { container } = render(<About {...props} />);
    expect(container).toMatchSnapshot();
  });

  test("should have render a single H1 tag", () => {
    const { container, getByText } = render(<About {...props} />);
    expect(getByText("About")).toBeInTheDocument();
    expect(container.getElementsByTagName("h1").length).toBe(1);
  });

  test("should toggle heading class", () => {
    const { getByTestId } = render(<About {...props} />);
    const heading = getByTestId("about-heading");
    expect(heading.className).toBe("heading--alt");
    userEvent.click(heading);
    expect(heading.className).toBe("heading");
  });

  test("should allow user to enter text value", () => {
    const { getByTestId } = render(<About {...props} />);
    const nameInput = getByTestId("about-input");
    const testString = "Hello, World!";
    userEvent.type(nameInput, testString);
    expect(props.dispatchUpdateName).toHaveBeenCalledTimes(testString.length);
    expect(output.value).toBe("Hello, World!");
  });
});
