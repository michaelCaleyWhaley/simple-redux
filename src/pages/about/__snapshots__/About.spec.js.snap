// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about page should render About component 1`] = `
<div>
  <h1
    class="heading--alt"
    data-testid="about-heading"
  >
    About
  </h1>
  <form>
    <label>
      Name:
      <input
        data-testid="about-input"
        id="name"
        name="name"
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Email:
      <input
        id="email"
        name="email"
        type="text"
        value=""
      />
    </label>
  </form>
  <button
    type="button"
  >
    Back
  </button>
</div>
`;
